servers/src/grin/sync/body_sync.rs

```diff
@@ -93,14 +93,25 @@ impl BodySync {
 		let head = self.chain.head()?;
 
 		// Find connected peers with strictly greater difficulty than us.
-		let peers: Vec<_> = self
-			.peers
-			.iter()
-			.outbound()
-			.with_difficulty(|x| x > head.total_difficulty)
-			.connected()
-			.into_iter()
-			.collect();
+		let peers_iter = || {
+			self.peers
+				.iter()
+				.with_difficulty(|x| x > head.total_difficulty)
+				.connected()
+		};
+
+		// We prefer outbound peers with greater difficulty.
+		let mut peers: Vec<_> = peers_iter().outbound().into_iter().collect();
+		if peers.is_empty() {
+			warn!("no outbound peers with more work, considering inbound");
+			peers = peers_iter().inbound().into_iter().collect();
+		}
+
+		// If we have no peers (outbound or inbound) then we are done for now.
+		if peers.is_empty() {
+			warn!("no peers (inbound or outbound) with more work");
```

Comment:
Nitpicking but why `warn` here. This is going to happen all the time no?

Reply:
Let me think about this a bit - it only happens once per actual transition out of "sync mode" as we only run body sync when we are actively syncing.